### PR TITLE
Task/timestamp process plugin ngsi2

### PR DESCRIPTION
**Pull request by dcalvoalonso**
This PR aims to add NGSIv2 support to timestamp process plugin.

Following the philosophy of https://github.com/dcalvoalonso/iotagent-node-lib/tree/task/activeAttsNgsi2, we have created specific functions for NGSIv2 in lib/plugins/pluginUtils.js and lib/plugins/timestampProcessPlugin.js. A unit test file has been also added test/unit/ngsiv2/plugins/timestamp-processing-plugin_test.js

grunt test and grunt lint are passed.

### CODE REVIEW

**Comment by dcalvoalonso** [comment]
New PR for timestamp process plugin: https://github.com/dcalvoalonso/iotagent-node-lib/pull/2

File: test/unit/ngsiv2/examples/contextRequests/updateContextProcessTimestamp1.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
+{
```
Comment (fgalan) [inline test/unit/ngsiv2/examples/contextRequests/updateContextProcessTimestamp1.json]: updateContextProcessTimestamp1.json -> updateContextProcessTimestamp.json ?

File: lib/plugins/pluginUtils.js [hunk 2]
```diff
@@ -173,3 +173,5 @@ exports.createProcessAttribute = createProcessAttribute;
 exports.createUpdateFilter = createUpdateFilter;
 exports.createQueryFilter = createQueryFilter;
 exports.identityFilter = identityFilter;
+exports.createNgsi2Entity = createNgsi2Entity;
```
Comment (fgalan) [inline lib/plugins/pluginUtils.js:176]: Following the philosophy used by PR #1, this PR should include this in the CNR file:

```
- Add: NGSIv2 support for Timestamp Processing plugin, through configuration (#527)
```

**Comment by fgalan** [comment]
Thanks for the contribution! Only minor comments in the review

File: test/unit/ngsiv2/examples/contextRequests/updateContextProcessTimestamp1.json (hunk 1 above)
Comment (dcalvoalonso) [reply test/unit/ngsiv2/examples/contextRequests/updateContextProcessTimestamp1.json]: Fixed in 50dec0c8461814b2cd32423764049e738f156571

File: lib/plugins/pluginUtils.js (hunk 2 above)
Comment (dcalvoalonso) [reply lib/plugins/pluginUtils.js:176]: Fixed in 50dec0c8461814b2cd32423764049e738f156571

File: test/unit/ngsiv2/examples/contextRequests/updateContextProcessTimestamp.json [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+        "value": "2016-05-30T16:25:22.304Z"
+      }
+    }
+  },
+  "TimeInstant":
+  {
+    "type": "DateTime",
+    "value": "2016-05-30T16:25:22.304Z"
+  }
+}
```
Comment (AlvaroVega) [inline test/unit/ngsiv2/examples/contextRequests/updateContextProcessTimestamp.json]: newline expected here

**Comment by AlvaroVega** [comment]
LGTM, wait for @fgalan for final LGTM before merge, pls

File: test/unit/ngsiv2/examples/contextRequests/updateContextProcessTimestamp.json (hunk 3 above)
Comment (dcalvoalonso) [reply test/unit/ngsiv2/examples/contextRequests/updateContextProcessTimestamp.json]: Fixed in 48e3530f18fee235e67e9cdc53fdf992b72e69a7

**Comment by dcalvoalonso** [comment]
No problem, once @fgalan gives his LGTM  I will do the Squash and merge

**Comment by fgalan** [comment]
LGTM. The PR is ready for "Squash&merge". Thanks!

**Comment by dcalvoalonso** [comment]
Thanks for your comments @fgalan and @AlvaroVega

**Pull request merged**